src/workerd/jsg/ser.c++

```diff
@@ -7,15 +7,25 @@
 
 namespace workerd::jsg {
 
-Serializer::ExternalHandler::~ExternalHandler() noexcept(false) {}
+void Serializer::ExternalHandler::serializeFunction(
+    jsg::Lock& js, jsg::Serializer& serializer, v8::Local<v8::Function> func) {
+  JSG_FAIL_REQUIRE(DOMDataCloneError, func, " could not be cloned.");
+}
 
-Serializer::Serializer(Lock& js, kj::Maybe<Options> maybeOptions)
-    : ser(js.v8Isolate, this) {
+Serializer::Serializer(Lock& js, Options options)
+    : externalHandler(options.externalHandler),
+      treatClassInstancesAsPlainObjects(options.treatClassInstancesAsPlainObjects),
+      ser(js.v8Isolate, this) {
 #ifdef KJ_DEBUG
   kj::requireOnStack(this, "jsg::Serializer must be allocated on the stack");
 #endif
-  auto options = kj::mv(maybeOptions).orDefault({});
-  externalHandler = options.externalHandler;
+  if (!treatClassInstancesAsPlainObjects) {
+    prototypeOfObject = js.obj().getPrototype();
+  }
+  if (externalHandler != kj::none) {
```

Comment:
Non-blocking for now, but I wonder if there shouldn't be a separate config `Option` for handling functions this way. I imagine we might want to use an external handler for `structuredClone` cases where we may not actually want to handle functions this way.

Reply:
The default implementation of `ExternalHandler::serializeFunction()` produces behavior identical to if you didn't enable treating functions as host objects, so I think a separate option isn't needed. You just don't override the method if you don't want functions.

Reply:
Yeah, I spotted that which is why I definitely don't think this is blocking. I'd just prefer not to engage any of this at all in cases where it's not needed. But, definitely not critical in any way.